src/main/java/org/ngafid/flights/process/CSVFileProcessor.java

```diff
@@ -46,105 +54,408 @@ private CSVFileProcessor(Connection connection, byte[] bytes, String filename, P
-                    columns.get(i).add(row[i]);
-                validRows += 1;
+            rows = filterValidRows(rows, headers);
+
+            boolean isG5FlightRecorder = isG5FlightRecorder(headerLines, firstRow);
+
+            // Library for mapping lat/long to timezones. Time zone for the US including Alaska and Hawaii.
+            TimeZoneMap timeZoneMap = null;
+            if (isG5FlightRecorder) {
+                timeZoneMap = TimeZoneMap.forRegion(18.91, -179.15, 71.538800, -66.93457);
+                rows = filterValidRowsG5(rows);
             }
 
-            // If we detect an invalid row before the last row of the file, or there are no valid rows.
-            if (validRows < Math.max(columns.size() - 2, 0)) {
-                throw new FatalFlightFileException(
-                        "Flight file has 0 valid rows - something serious is wrong with the format.");
+            List<Integer> splitIndices = splitCSVIntoFlightIndices(rows, 5);
+            addDerivedFileToUploadFolder(splitIndices, rows, headerLines, filename);
+
+            // Process each flight by splitting the series based on the indices
+            for (int i = 0; i < splitIndices.size(); i++) {
+                int fromIndex = splitIndices.get(i);
+                int toIndex = (i == splitIndices.size() - 1) ? rows.size() : splitIndices.get(i + 1);
+                List<String[]> flightRows = rows.subList(fromIndex, toIndex);
+
+                // Create columns for each flight segment
+                ArrayList<ArrayList<String>> columns = new ArrayList<>();
+                for (int j = 0; j < firstRow.length; j++) {
+                    columns.add(new ArrayList<>());
+                }
+                // Populate columns with data from the flightRows
+                for (String[] row : flightRows) {
+                    if (row.length < firstRow.length) {
+                        break;
+                    }
+                    for (int j = 0; j < row.length; j++) {
+                        columns.get(j).add(row[j]);
+                    }
+                }
+                // G5 doesn't have local date time and utcOffset column, we calculate them here
+                TimeUtils.LocalDateTimeResult localDateTimeResult = null;
+                if(isG5FlightRecorder) {
+                    StringTimeSeries utcDateSeries = new StringTimeSeries(headers.get(0), dataTypes.get(0), columns.get(0));
+                    StringTimeSeries utcTimeSeries = new StringTimeSeries(headers.get(1), dataTypes.get(1), columns.get(1));
+                    DoubleTimeSeries latitudeTimeSeries = new DoubleTimeSeries(headers.get(4), dataTypes.get(4), columns.get(4));
+                    DoubleTimeSeries longitudeTimeSeries = new DoubleTimeSeries(headers.get(5), dataTypes.get(5), columns.get(5));
+                    localDateTimeResult = TimeUtils.calculateLocalDateTimeFromTimeSeries(timeZoneMap, utcDateSeries,utcTimeSeries,latitudeTimeSeries,longitudeTimeSeries);
```

Comment:
I would recommend not assuming the order of the columns will alwys be the same for the g5 recorder.

Also, I think you can preserve the original column names using this aliasing system:
https://github.com/travisdesell/ngafid2.0/blob/josh/refactor/src/main/java/org/ngafid/flights/process/FlightBuilder.java#L137


Basically, you will alias g5 column names to the normalized names we use (e.g. whatever g5 uses for lcl date will be aliased to Lcl Date). If this isn't sufficient, you may want to move this to its own processing step that outputs Lcl Date etc. columns and subclass the CSVFlightBuilder to add that additional processing step.